server/routes/auth.js

```diff
@@ -81,44 +43,7 @@ authRouter.get('/facebook', passport.authenticate('facebook', {
 
 authRouter.get('/facebook/callback', (req, res, next) => {
   passport.authenticate('facebook', (err, user) => {
-    if (err) {
-      return res.status(400).send({
-        status: 400,
-        success: false,
-        message: 'TokenError: The token has already been used'
-      });
-    }
-    if (!user) {
-      return res.status(401).send({
-        status: 401,
-        success: false,
-        message: 'Facebook Authentication Failed'
-      });
-    }
-    req.login(user, (loginErr) => {
-      if (loginErr) {
-        return next(loginErr);
-      }
-      const token = JwtHelper.createToken({
-        user: {
-          id: user.id,
-          username: user.username,
-          email: user.email
-        }
-      }, '24h');
-      return res.status(200).send({
-        status: 200,
-        success: true,
-        message: 'Facebook Authentication Succeeded',
-        user: {
-          username: user.username,
-          firstName: user.firstName,
-          lastName: user.lastName,
-          email: user.email,
-          token,
-        },
-      });
-    });
+    AuthController.finishSocialLogin(err, user, res, 'Facebook');
```

Comment:
Weldone on this! It is really neat, I am only thinking if you had raised a bug fix to refactor the code and write test, it would have been the best practice. But you have really done well